src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.addAnimal(this);
+  }
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    const index = Animal.alive.indexOf(animal);
+
+    Animal.alive.splice(index, 1);
```

Comment:
`[CODE PERFOMANCE] - Try to avoid nesting loops (indexOf + splice), use filter instead`